internal/cli/workspace_list.go

```diff
@@ -0,0 +1,135 @@
+package cli
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/olekukonko/tablewriter"
+	"github.com/posener/complete"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/hashicorp/waypoint-plugin-sdk/terminal"
+	"github.com/hashicorp/waypoint/internal/clierrors"
+	"github.com/hashicorp/waypoint/internal/pkg/flag"
+	"github.com/hashicorp/waypoint/internal/server/gen"
+	pb "github.com/hashicorp/waypoint/internal/server/gen"
+)
+
+type WorkspaceListCommand struct {
+	*baseCommand
+}
+
+func (c *WorkspaceListCommand) Run(args []string) int {
+	// Initialize. If we fail, we just exit since Init handles the UI.
+	if err := c.Init(
+		WithArgs(args),
+		WithFlags(c.Flags()),
+		WithNoConfig(),
+	); err != nil {
+		return 1
+	}
+
+	resp, err := c.project.Client().ListWorkspaces(c.Ctx, &pb.ListWorkspacesRequest{
+		Scope: &pb.ListWorkspacesRequest_Global{},
+	})
+	if err != nil {
+		c.ui.Output(clierrors.Humanize(err), terminal.WithErrorStyle())
+		return 1
+	}
+
+	var result []string
+	for _, p := range resp.Workspaces {
+		result = append(result, p.Name)
+	}
+
+	if len(result) == 0 {
+		c.ui.Output("No workspaces found.")
+		return 0
+	}
+	sort.Strings(result)
+
+	// Get our direct stdout handle because we're going to be writing colors
+	// and want color detection to work.
+	out, _, err := c.ui.OutputWriters()
+	if err != nil {
+		c.ui.Output(clierrors.Humanize(err), terminal.WithErrorStyle())
+		return 1
+	}
+
+	table := tablewriter.NewWriter(out)
+	table.SetHeader([]string{"Name", "Projects"})
+	table.SetBorder(false)
```

Comment:
Not sure if you knew, but we have a built in table writer within UI which would help keep our `waypoint * list` functions looking the same.

You can initialize the table from the SDK terminal package

```
tbl := terminal.NewTable(headers...)
```

And then build up the columns just as you're already doing with `tbl.Rich`, and print using the UI package with `c.ui.Table(tbl)`

Reply:
Thanks! I refactored that in 1880c9ec

Reply:
Thanks! I refactored that in 1880c9e